core/includes/bootstrap.inc

```diff
@@ -611,6 +611,7 @@ function find_conf_path($http_host, $script_name, $require_settings = TRUE) {
 
   if ($sites) {
     $uri = explode('/', $script_name);
+    $uri[0] = ($uri[0] == '.') ? '' : $uri[0];
```

Comment:
What does this change accomplish?

Reply:
So the `$uri` variable in the line above has the following array when the UI installer is used:

```
0 => '',
1 => 'core',
2 => 'install.php',
```

When the `install.sh` script is used though, the array is:

```
0 => '.',
1 => 'core',
2 => 'scripts',
3 => 'install.sh',
```

The following `$dir` variable is then set to the following values (during the loop), looking for a matching domain entry in the `sites.php` file...

UI installer:

```
example.com.core
com.core
example.com
com
```

And the `install.sh` script:

```
example.com..core.scripts
com..core.scripts
example.com..core
com..core
example.com.
com.
```

As you can see, the UI installer will eventually find and match `example.com`, whereas the `install.sh` script won't (the closest it gets is `example.com.` (note the end `.`)).

So this line of code addition in the PR basically checks to see if the first `$uri` array item is a single `.` and, if so, replaces it with an empty string. This makes it possible to properly find a matching domain in `sites.php`.

Reply:
Huh, that seems dumb that directories like "core" or "core/scripts" get added to the search keys in the first place. But your explanation makes sense to get consistent results. And that would be an issue with the sites.php parsing, rather than the install.sh script.